addons/payment/wizards/payment_link_wizard.py

```diff
@@ -62,17 +63,60 @@ def _compute_values(self):
     @api.depends('res_model', 'res_id')
     def _compute_company(self):
         for link in self:
-            record = self.env[link.res_model].browse(link.res_id)
-            link.company_id = record.company_id if 'company_id' in record else False
+            related_document = self.env[link.res_model].browse(link.res_id)
+            link.company_id = related_document.company_id
+
+    @api.depends('payment_acquirer')
+    def _compute_has_multiple_acquirer(self):
+        for link in self:
+            if len(self._get_payment_acquirer_available()) > 2:
+                link.has_multiple_acquirer = True
+            else:
+                link.has_multiple_acquirer = False
+
+    def _get_payment_acquirer_available(self):
+        """ Specify available acquirers in the selection field.
+
+        :return: a list of tuples, the first one being the 'all' one, and then
+        one tuple for each available acquirer, with state enabled or test.
+        :rtype: list(tuple(id, name), ...)
+        """
+        defaults = self.default_get(['res_model', 'res_id'])
+        selection = [('all', "All")]
+        res_model, res_id = defaults['res_model'], defaults['res_id']
+        if res_model and res_id:
+            # At module install, the selection method is called
+            # but the document context isn't specified.
+            related_document = self.env[res_model].browse(res_id)
+            company_id = related_document.company_id
+            domain = [
+                ('company_id', '=', company_id.id),
+                ('state', 'in', ['enabled', 'test']),
+            ]
+            selection.extend(self.env['payment.acquirer'].search(domain).name_get())
+        return selection
+
+    @api.onchange('amount', 'description')
+    def _onchange_amount(self):
+        if float_compare(
+            self.amount_max, self.amount,
+            precision_rounding=self.currency_id.rounding or 0.01
+        ) == -1:
+            raise ValidationError(_("Please set an amount smaller than %s.") % (self.amount_max))
+        if self.amount <= 0:
+            raise ValidationError(_("The value of the payment amount must be positive."))
 
     def _generate_link(self):
         for payment_link in self:
             related_document = self.env[payment_link.res_model].browse(payment_link.res_id)
-            base_url = related_document.get_base_url()  # Don't generate links for the wrong website
+            base_url = related_document.get_base_url() # Don't generate links for the wrong website
```

Comment:
```suggestion
            base_url = related_document.get_base_url()  # Don't generate links for the wrong website
```
Keep the 2 space chars before an inline comment in Python.